upgrades/upgrade_testing_framework/steps/step_create_source_snapshot.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+import upgrade_testing_framework.core.shell_interactions as shell
+
+class CreateSourceSnapshot(FrameworkStep):
+    """
+    This step creates the snapshot on the source cluster.
+    """
+
+    def _run(self):
+        """
+        The contents of this are quick/dirty.  Instead of raw curl commands we should probably be using the Python SDK
+        for Elasticsearch/OpenSearch.  At the very least, we should wrap the curl commands in our own pseudo-client.
```

Comment:
minor: want to make/link to a gh issue for this task? I agree that at least a pseudo-client sounds like a necessary approach and I'd like to have it on our radar.

Reply:
Already made and in progress - https://github.com/opensearch-project/opensearch-migrations/issues/50